src/reconcile/rec_write.c

```diff
@@ -382,6 +382,15 @@ __wt_reconcile(WT_SESSION_IMPL *session,
 	WT_ASSERT(session, WT_TXNID_LE(mod->last_oldest_id, oldest_id));
 	mod->last_oldest_id = oldest_id;
 
+	/*
+	 * Evicting in-memory uses the update/restore mechanisms.
+	 * The update/restore mechanisms use disk images.
+	 */
+	if (LF_ISSET(WT_EVICT_IN_MEMORY))
+		LF_SET(WT_EVICT_UPDATE_RESTORE);
+	if (LF_ISSET(WT_EVICT_UPDATE_RESTORE))
+		LF_SET(WT_EVICT_SCRUB);
```

Comment:
I don't like changing the flags to change here -- I would like the behavior to be fully specified by the eviction layer.